sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -592,6 +594,48 @@ def __init__(self, special_parameter):
                                   check_estimator, MyEstimator)
 
 
+class LRXFailTagButCheckPasses(LogisticRegression):
+    def _more_tags(self):
+        return {"_xfail_checks": {
+            # check_complex_data actually passes for this estimator
+            "check_complex_data": "This is a bad check"}
+        }
+
+
+def test_check_estimator_xfail_tag_warns():
+    # When a _xfail_checks check passes, warns stating that the test
+    # passes and can be removed from the tag
+
+    estimators = [LRXFailTagButCheckPasses(), LRXFailTagButCheckPasses]
+
+    for estimator in estimators:
+        assert_warns_message(FutureWarning,
+                             "LRXFailTagButCheckPasses:check_complex_data "
+                             "passed, it can be removed from the "
+                             "_xfail_check tag",
+                             check_estimator, estimator)
+
+
+class LRXFailDoesNotRaiseOnComplex(LRXFailTagButCheckPasses):
+    def fit(self, X, y):
+        # do not raise error for complex check
+        try:
+            return super().fit(X, y)
+        except ValueError as e:
+            if "Complex data not supported" not in str(e):
+                raise
+        return self
+
+
+def test_check_estimator_xfail_tag_skips():
+    # skips check_complex_data based on _xfail_checks
```

Comment:
Can't we just check on an estimator with a properly set `xfail_tags` instead, like the Dummy one?

I don't understand why we need to create a new estimator here

Reply:
I was thinking that if the estimator gets fixed and the `xfail_tags` gets removed this test would start failing.

I guess we can use Dummy for now.